mlir/lib/Ion/Transforms/CMakeLists.txt

```diff
@@ -7,10 +7,39 @@ file(GLOB SRC
 
 get_property(dialect_libs GLOBAL PROPERTY MLIR_DIALECT_LIBS)
 get_property(conversion_libs GLOBAL PROPERTY MLIR_CONVERSION_LIBS)
+
+
+
+
+find_package(Python 3
+    REQUIRED COMPONENTS Interpreter Development.Module
+    OPTIONAL_COMPONENTS Development.SABIModule)
+
+# nanobind suggests including these lines to configure CMake to perform an optimized release build
+# by default unless another build type is specified. Without this addition, binding code may run
+# slowly and produce large binaries.
+# See https://nanobind.readthedocs.io/en/latest/building.html#preliminaries
+
+#if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+#    set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
+#    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+#endif()
+
+# Detect the installed nanobind package and import it into CMake
+execute_process(
+    COMMAND "${Python_EXECUTABLE}" -m nanobind --cmake_dir
+    OUTPUT_VARIABLE nanobind_ROOT OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+find_package(nanobind CONFIG REQUIRED)
+
+
+include_directories(${Python3_INCLUDE_DIRS})
+
 set(LIBS
     ${dialect_libs}
     ${conversion_libs}
     MLIRIon
+    ${Python3_LIBRARIES}
```

Comment:
I guess here when compiling the new header for the pass I would need to link in the nanobind/python libraries, but I am unsure how to do that.